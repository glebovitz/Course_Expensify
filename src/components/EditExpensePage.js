import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) =>  {
    const props = this.props;
    props.editExpense(props.expense.id, expense);
    props.history.push("/");
  };
  onClick = () => {
    const props = this.props;
    props.removeExpense({ id: props.expense.id });
    props.history.push("/");
  };
  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button
          onClick={this.onClick}
        >
          Remove
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id == props.match.params.id)
  }
}

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (data) => dispatch(removeExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);