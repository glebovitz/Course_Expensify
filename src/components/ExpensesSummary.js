import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
  return (
    <div>
      ExpenseTotal: {numeral(selectExpensesTotal(props.expenses)/100).format('$0,0.0')}
      {' - '}
      ExpenseCount: {props.expenses.length}
    </div>)
};

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);
