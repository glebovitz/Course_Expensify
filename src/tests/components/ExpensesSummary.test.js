import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test ('should render ExpensesSummary with 3 expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary 
      expensesTotal={114200}
      expensesCount={3}
    />);
  expect(wrapper).toMatchSnapshot();
});

test ('should render ExpensesSummary with multiple expense', () => {
  const wrapper = shallow(
    <ExpensesSummary 
      expensesTotal={109500}
      expensesCount={1}
    />);
  expect(wrapper).toMatchSnapshot();
});
