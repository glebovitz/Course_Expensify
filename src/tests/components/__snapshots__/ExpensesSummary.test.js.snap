// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 3 expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,142.0
  </h1>
</div>
`;

exports[`should render ExpensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1,095.0
  </h1>
</div>
`;
